backend/src/test/java/com/redhat/cloud/notifications/routers/internal/errata/ErrataUserPreferencesMigrationResourceTest.java

```diff
@@ -0,0 +1,174 @@
+import java.net.URL;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Set;
+
+import static com.redhat.cloud.notifications.routers.internal.errata.ErrataUserPreferencesMigrationResource.EVENT_TYPE_NAME_BUGFIX;
+import static com.redhat.cloud.notifications.routers.internal.errata.ErrataUserPreferencesMigrationResource.EVENT_TYPE_NAME_ENHANCEMENT;
+import static com.redhat.cloud.notifications.routers.internal.errata.ErrataUserPreferencesMigrationResource.EVENT_TYPE_NAME_SECURITY;
+import static io.restassured.RestAssured.given;
+
+@QuarkusTest
+@QuarkusTestResource(TestLifecycleManager.class)
+public class ErrataUserPreferencesMigrationResourceTest extends DbIsolatedTest {
+    @InjectSpy
+    ErrataMigrationRepository errataMigrationRepository;
+
+    @Inject
+    ResourceHelpers resourceHelpers;
+
+    @ConfigProperty(name = "internal.admin-role")
+    String adminRole;
+
+    @Inject
+    SubscriptionRepository subscriptionRepository;
+
+    /**
+     * Tests that when there is an unsupported Errata event type the call just
+     * fails.
+     * @throws URISyntaxException if the JSON file's URL is not properly built.
+     */
+    @Test
+    void testUnsupportedEventType() throws URISyntaxException {
+        // Simulate that the repository returns a non-Errata event type.
+        final EventType notAnErrataEventType = new EventType();
+        notAnErrataEventType.setName("not-an-errata-event-type");
+
+        Mockito.when(this.errataMigrationRepository.findErrataEventTypes()).thenReturn(List.of(notAnErrataEventType));
+
+        // Load the file input for the request.
+        final URL jsonResourceUrl = this.getClass().getResource("/errata/subscriptions/errata_subscriptions.json");
+        if (jsonResourceUrl == null) {
+            Assertions.fail("The path of the JSON test file is incorrect");
+        }
+
+        final File file = Paths.get(jsonResourceUrl.toURI()).toFile();
+
+        // Call the endpoint under test.
+        final String response = given()
+            .basePath(Constants.API_INTERNAL)
+            .header(TestHelpers.createTurnpikeIdentityHeader("user", this.adminRole))
+            .when()
+            .contentType(MediaType.MULTIPART_FORM_DATA)
+            .multiPart("jsonFile", file)
+            .post("/team-nado/migrate/json")
+            .then()
+            .statusCode(HttpStatus.SC_INTERNAL_SERVER_ERROR)
+            .extract()
+            .asString();
+
+        Assertions.assertEquals("A non errata event type was detected. Nothing was executed.", response, "unexpected response received for when an unexpected event type gets fetched from the database");
+    }
+
+
+     /**
+     * Tests that uploading a JSON file with the Errata subscriptions creates
+     * the corresponding subscriptions in our database.
+     * @throws URISyntaxException if the JSON file's URL is not properly built.
+     */
+    @Test
+    void testJsonMigration() throws URISyntaxException {
```

Comment:
Could you please set `BATCH_SIZE` to `2` from the test so that the whole `while` loop from `ErrataUserPreferencesMigrationResource` is tested?

Reply:
Yes, and I've added an environment variable so that we can configure it in case we need to reduce the batch size due to memory constraints.